web3/_utils/proof.py

```diff
@@ -0,0 +1,133 @@
+from eth_utils import (
+    keccak,
+)
+import rlp
+from rlp.sedes import (
+    Binary,
+    big_endian_int,
+)
+
+from web3._utils.encoding import (
+    pad_bytes,
+    pad_hex,
+)
+
+
+def storage_position(map_key_hex, position_hex):
+    '''
+    https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_getstorageat
+    Returns the patricia trie key (position) of a mapping's key.
+    This helper function is used for formating arguments to eth_getStorageAt
+    and eth_getProof.
+    @param map_key_hex is the key stored in a map
+    @param position_hex is the position where the mapping variable is defined
+        in the contract
+    '''
+    key = pad_hex(map_key_hex, 256)[2:] + pad_hex(position_hex, 256)[2:]
+    return keccak(bytes.fromhex(key)).hex()
+
+
+nibble_to_number = {}
+for i, c in enumerate('0123456789abcdef'):
+    nibble_to_number[c] = i
+
+
+class _Account(rlp.Serializable):
+    fields = [
+        ('nonce', big_endian_int),
+        ('balance', big_endian_int),
+        ('storage', Binary.fixed_length(32, allow_empty=True)),
+        ('code_hash', Binary.fixed_length(32))
+    ]
+
+
+def verify_eth_getProof(proof, root):
```

Comment:
Since the `getProof` API doesnt return a bool for whether or not a value is included (inclusion or exclusion should be verified for each key being proved), verifying the getProof result means verifying all values are included or all are not included. But we can't verify a mixture of inclusion and exclusion storage proofs. In this case `verify_eth_getProof` verifies all keys are included. possible rename to `verify_getProof_inclusion` ?

Reply:
Do you think this should be a publicly-used API? If so, it shouldn't be in `_utils`. If not, it's not *as* important to find an ideal name. `inclusion` does seem a bit better, at first glance.